intervals/main_objs.py

```diff
@@ -1576,9 +1706,55 @@ def _split_by_threshold(seq, max_diff=70):
 
         yield part
 
+    # July 2022 added to check for overlap
+    def _dur_ngram_helper(df, ng_durs):
+        """
+        This helper is used by the presentation type classifier to check
+        total duration of each ngram in a given set of soggetti for a
+        presentation type.
+
+        Note that it relies on the ng_dur dataframe that appears in the
+        presentation type classifier itself:
+
+        ng_durs = self.durations(df=entries)
+
+        These in turn are saved as a string to check for overlaps among entries.
+        """
+        dur_list = []
+        l = list(zip(df["Offsets"], df["Voices"]))
+        for pair in l:
```

Comment:
There is also room for optimization here, so maybe Oleh could profile some of the slower functions and see if this is a bottleneck. Not a big issue though, since it's just about speed and not the actual result.